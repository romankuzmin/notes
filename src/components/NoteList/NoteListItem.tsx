import { ListItem, ListItemText } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import React, { FC, forwardRef, memo, useMemo } from 'react';
import { Link, LinkProps as RouterLinkProps } from 'react-router-dom';

const useStyles = makeStyles({
    root: {
        listStyle: 'none',
    },
});

type NoteProps = {
    id: string;
    title: string;
};

const NoteListItem: FC<NoteProps> = ({ id, title }) => {
    const classes = useStyles();

    const CustomLink = useMemo(
        () =>
            // eslint-disable-next-line react/display-name
            forwardRef<never, Omit<RouterLinkProps, 'to'>>((linkProps, ref) => (
                <Link to={`/${id}`} {...linkProps} ref={ref} />
            )),
        [id],
    );

    return (
        <div className={classes.root}>
            <ListItem button={true} component={CustomLink}>
                <ListItemText primary={title} />
            </ListItem>
        </div>
    );
};

export default memo(NoteListItem);
