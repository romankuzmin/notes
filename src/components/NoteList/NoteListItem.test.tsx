import React from 'react';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import '@testing-library/jest-dom/extend-expect';
import { Router } from 'react-router';

import NoteListItem from './NoteListItem';

test('Render default item', () => {
    const title = 'Note test';
    const history = createMemoryHistory();
    const { asFragment, getByRole } = render(
        <Router history={history}>
            <NoteListItem id="1" title={title} />
        </Router>,
    );

    expect(getByRole('button')).toHaveAttribute('href', '/1');
    expect(asFragment()).toMatchSnapshot();
});
