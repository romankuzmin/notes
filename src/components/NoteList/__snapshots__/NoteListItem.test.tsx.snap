// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render default item 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      href="/1"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Note test
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
</DocumentFragment>
`;
